// yiffy/build/src/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Yiffy = void 0;
const Constants_1 = require("./util/Constants");
const Animals_1 = __importDefault(require("./Endpoints/Animals"));
const Furry_1 = __importDefault(require("./Endpoints/Furry"));
class Yiffy {
    options;
    _animals;
    _furry;
    constructor(d) {
        if (!d)
            d = {};
        this.options = {
            userAgent: d.userAgent || Constants_1.USER_AGENT,
            apiKey: d.apiKey || "",
            debug: d.debug || (() => null),
            baseURL: d.baseURL || Constants_1.API_URL,
            host: d.host || Constants_1.API_HOST
        };
    }
    get animals() { return this._animals || (this._animals = new Animals_1.default(this.options)); }
    get furry() { return this._furry || (this._furry = new Furry_1.default(this.options)); }
    get debug() { return this.options.debug; }
}
exports.Yiffy = Yiffy;
exports.default = Yiffy;
module.exports = Yiffy;
