// yiffy/build/src/util/Constants.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.USER_AGENT = exports.API_HEADERS = exports.API_HOST = exports.API_URL = exports.API_VERSION = void 0;
const package_json_1 = __importDefault(require("../../package.json"));
exports.API_VERSION = "V2";
exports.API_URL = `https://${exports.API_VERSION.toLowerCase()}.yiff.rest`;
exports.API_HOST = `${exports.API_VERSION.toLowerCase()}.yiff.rest`;
exports.API_HEADERS = {
    artists: "X-Yiffy-Artist",
    source: "X-Yiffy-Source",
    width: "X-Yiffy-Image-Width",
    height: "X-Yiffy-Image-Height",
    url: "X-Yiffy-Image-URL",
    shortURL: "X-Yiffy-Short-URL",
    reportURL: "X-Yiffy-Report-URL",
    type: "X-Yiffy-Image-Type",
    name: "X-Yiffy-Image-Name",
    size: "X-Yiffy-Image-Size",
    ext: "X-Yiffy-Image-Extension"
};
exports.USER_AGENT = `Yiffy/${package_json_1.default.version} (https://${exports.API_VERSION.toLowerCase()}.yiff.rest, https://github.com/DonovanDMC/Yiffy)`;
