// yiffy/build/src/util/get.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const Constants_1 = require("./Constants");
const https = __importStar(require("https"));
const url_1 = require("url");
async function get(url, ua, auth, host) {
    return new Promise((a, b) => {
        const u = new url_1.URL(url);
        https
            .get({
            host: u.host,
            path: `${u.pathname}${u.search}`,
            port: url.includes("https") ? 443 : 80,
            method: "GET",
            headers: {
                "User-Agent": ua,
                "Host": host || Constants_1.API_HOST,
                ...(auth ? ({
                    Authorization: auth
                }) : ({}))
            }
        }, (res) => {
            const d = [];
            res
                .on("data", d.push.bind(d))
                .on("error", (err) => b(err))
                .on("end", () => a({
                headers: res.headers,
                body: Buffer.concat(d),
                statusCode: res.statusCode,
                statusMessage: res.statusMessage
            }));
        })
            .end();
    });
}
exports.default = get;
